web/src/components/PinList.vue

```diff
@@ -56,11 +56,40 @@ import {
 } from '@vue/composition-api';
 import store from '../store';
 import { Pin } from '../generatedTypes/AtlascopeTypes';
+import { Point, postGisToPoint } from '../utilities/utiltyFunctions';
 
 export default defineComponent({
   setup() {
     const pins: Ref<Pin[]> = computed(() => store.state.currentPins);
+    const bounds = computed(() => store.state.currentBounds);
+    const zoom = computed(() => store.state.zoomLevel);
     const rootDataset = computed(() => store.state.rootDataset);
+
+    function isInBounds(pin: Pin): boolean {
+      const point: Point = postGisToPoint(pin.location) || { x: 0, y: 0 };
+      const containedInMapBounds = (point.x < bounds.value.right
+        && point.x > bounds.value.left
+        && point.y > bounds.value.top
+        && point.y < bounds.value.bottom);
+      const pinMaxZoom = pin.maximum_zoom || 40;
+      const pinMinZoom = pin.minimum_zoom || 0;
+      const visibleAtCurrentZoom = (
+        pinMinZoom - 2 <= zoom.value
+        && pinMaxZoom + 2 >= zoom.value
```

Comment:
For readability, you may want to invert this test:

```suggestion
        && zoom.value <= pinMaxZoom + 2
```